components/tools/OmeroPy/src/omero/plugins/chgrp.py

```diff
@@ -84,16 +84,10 @@ def _process_request(self, req, args, client):
 
     def create_error_report(self, rsp):
         import omero.cmd
-        if isinstance(rsp, omero.cmd.GraphConstraintERR):
-            if "Fileset" in rsp.constraints:
-                fileset = rsp.constraints.get("Fileset")
-                return "You cannot move part of fileset %s; only complete" \
-                    " filesets can be moved to another group.\n" % \
-                    ", ".join(str(x) for x in fileset)
-            else:
-                return super(ChgrpControl, self).create_error_report(rsp)
-        else:
-            return super(ChgrpControl, self).create_error_report(rsp)
+        if isinstance(rsp, omero.cmd.GraphException):
+            return 'Chgrp failed: %s' % rsp.message
```

Comment:
The graph exception is now handled at the API level, I would argue there is no need to subclass `create_error_report()` anymore. Instead this line could be directly moved to `GraphControl.create_error_report()` so that other plugins like `delete` can also output the graph exception.
Ideally we could also add corresponding integration tests for `bin/omero delete` as part of this refactoring. I can open a PR against this branch to do so if you want.